File: applications/vehicle-sink/src/main/java/com/vmware/vehicle/sink/functions/VehicleRepositorySink.java
package com.vmware.vehicle.sink.functions;

import com.vmware.tanzu.data.IoT.vehicles.domains.Vehicle;
import com.vmware.vehicle.sink.repository.VehicleRepository;
import org.springframework.stereotype.Component;

import java.util.function.Consumer;

@Component
public class VehicleRepositorySink implements Consumer<Vehicle> {
    private final VehicleRepository repository;

    public VehicleRepositorySink(VehicleRepository repository) {
        this.repository = repository;
    }

    @Override
    public void accept(Vehicle vehicle) {
        repository.save(vehicle);
    }
}

File: components/iot-connected-vehicles-repository-sink/src/main/kotlin/com/vmware/tanzu/data/IoT/vehicles/sink/vehicle/VehicleRepositorySink.kt
package com.vmware.tanzu.data.IoT.vehicles.sink.vehicle

import com.vmware.tanzu.data.IoT.vehicles.domains.Vehicle
import com.vmware.tanzu.data.IoT.vehicles.repositories.VehicleRepository
import nyla.solutions.core.util.Debugger
import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Service
import java.lang.RuntimeException
import java.util.function.Consumer


/**
 * @author Gregory Green
 */
@Service
class VehicleRepositorySink(
    private val vehicleRepository: VehicleRepository
) : Consumer<Vehicle> {
    override fun accept(vehicleData: Vehicle) {
        try {
            vehicleRepository.save(vehicleData);
        }
        catch (e : RuntimeException)
        {
            println("ERROR: Exception $e Vehicle: $vehicleData TRACK: ${Debugger.stackTrace(e)}")
            throw e;
        }

    }
}